### PR TITLE
Adding reference for create cluster and cluster definition format

### PR DESCRIPTION
**Pull request by marians**
This adds the documentation needed to create and delete a cluster using `gsctl`.

**Note:** This should not be merged before that `gsctl` function is actually released.

See https://github.com/giantswarm/gsctl/pull/14

Related: https://github.com/giantswarm/docs/pull/199

### CODE REVIEW

**Comment by marians** [comment]
@puja108 Can you re-check, please?

File: content/reference/cluster-definition/index.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+title = "Cluster Definition YAML Reference"
+description = "Complete documentation of the Giant Swarm cluster definition YAML format"
+date = "2017-01-05"
+type = "page"
+weight = 10
++++
+
+# Cluster Definition YAML Reference
+
+The [`gsctl create cluster`](../gsctl/create-cluster/) command can consume YAML files conforming this specification to create new clusters.
```
Comment (JosephSalisbury) [inline content/reference/cluster-definition/index.md]: Tiny nitpick - it should be `conforming to this specification`

Comment (marians) [inline content/reference/cluster-definition/index.md]: Changed

**Comment by marians** [comment]
@puja108 @JosephSalisbury There have been a few changes since your reviews. Anyway I'd like to publish this, so I'll merge.

**Pull request merged**